In [2]:
# import neessary libraries
import pandas as pd
from pyampute.exploration.mcar_statistical_tests import MCARTest

In [3]:
# load data
df_satellite_to_append = []
df_solar_to_append = []
df_wind_to_append = []

for farm in [1, 2, 3, 4, 5]:
    satellite_to_append = pd.read_csv(f"data/solar/enlitia_satellite_{farm}_data.csv")
    solar_to_append = pd.read_csv(f"data/solar/enlitia_solar_farm_{farm}_data.csv")
    wind_to_append = pd.read_csv(f"data/wind/enlitia_wind_farm_{farm}_data.csv")
    
    df_satellite_to_append.append(satellite_to_append)
    df_solar_to_append.append(solar_to_append)
    df_wind_to_append.append(wind_to_append)

df_satellite = pd.concat(df_satellite_to_append, ignore_index=True)
df_solar = pd.concat(df_solar_to_append, ignore_index=True)
df_wind = pd.concat(df_wind_to_append, ignore_index=True)

print("Satellite DataFrame:\n", df_satellite)
print("Solar DataFrame:\n", df_solar)
print("Wind DataFrame:\n", df_wind)


Satellite DataFrame:
         satellite_id              read_at  global_tilted_irradiance  \
0                  1  2022-01-01 00:00:00                       0.0   
1                  1  2022-01-01 00:10:00                       0.0   
2                  1  2022-01-01 00:20:00                       0.0   
3                  1  2022-01-01 00:30:00                       0.0   
4                  1  2022-01-01 00:40:00                       0.0   
...              ...                  ...                       ...   
476007             5  2024-01-09 14:00:00                     119.0   
476008             5  2024-01-09 14:10:00                     121.0   
476009             5  2024-01-09 14:20:00                      82.0   
476010             5  2024-01-09 14:30:00                      49.0   
476011             5  2024-01-09 14:40:00                      21.0   

        global_horizontal_irradiance  temperature  cloud_cover  
0                                0.0         13.0          1

In [4]:
dfs_dict = {
    "Satellite" : df_satellite,
    "Solar": df_solar,
    "Wind" : df_wind
}

for key, value in dfs_dict.items():
    print(f"Information of {key} df")
    print(value.info(),"\n")

Information of Satellite df
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 476012 entries, 0 to 476011
Data columns (total 6 columns):
 #   Column                        Non-Null Count   Dtype  
---  ------                        --------------   -----  
 0   satellite_id                  476012 non-null  int64  
 1   read_at                       476012 non-null  object 
 2   global_tilted_irradiance      476012 non-null  float64
 3   global_horizontal_irradiance  476012 non-null  float64
 4   temperature                   476012 non-null  float64
 5   cloud_cover                   476012 non-null  float64
dtypes: float64(4), int64(1), object(1)
memory usage: 21.8+ MB
None 

Information of Solar df
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4081774 entries, 0 to 4081773
Data columns (total 8 columns):
 #   Column      Dtype  
---  ------      -----  
 0   asset_id    int64  
 1   read_at     object 
 2   ac_power    float64
 3   ac_voltage  float64
 4   ac_current  float64
 

In [5]:
for key, value in dfs_dict.items():
    print(f"NAs count for {key}")
    print(value.isna().sum(), "\n")

NAs count for Satellite
satellite_id                    0
read_at                         0
global_tilted_irradiance        0
global_horizontal_irradiance    0
temperature                     0
cloud_cover                     0
dtype: int64 

NAs count for Solar
asset_id          0
read_at           0
ac_power          0
ac_voltage    99722
ac_current    99722
dc_power          0
dc_voltage        0
dc_current        0
dtype: int64 

NAs count for Wind
asset_id                0
read_at                 0
wind_speed              0
wind_direction          0
power_average           0
rotation_average        0
exterior_temperature    1
nacelle_temperature     1
dtype: int64 



In [6]:
df_solar_only_na = df_solar[df_solar["ac_voltage"].isna()]
print(df_solar_only_na)
#df_solar_only_na.to_csv("outputs/solar_only_nas.csv")

         asset_id              read_at  ac_power  ac_voltage  ac_current  \
3109038       411  2022-01-01 07:40:00       0.0         NaN         NaN   
3109039       411  2022-01-01 07:50:00       0.0         NaN         NaN   
3109040       411  2022-01-01 08:00:00       0.0         NaN         NaN   
3109041       411  2022-01-01 08:10:00     259.0         NaN         NaN   
3109042       411  2022-01-01 08:20:00     531.0         NaN         NaN   
...           ...                  ...       ...         ...         ...   
3208755       412  2024-01-09 13:20:00    1857.0         NaN         NaN   
3208756       412  2024-01-09 13:30:00    2015.0         NaN         NaN   
3208757       412  2024-01-09 13:40:00    2114.0         NaN         NaN   
3208758       412  2024-01-09 13:50:00    1459.0         NaN         NaN   
3208759       412  2024-01-09 14:00:00    4352.0         NaN         NaN   

         dc_power  dc_voltage  dc_current  
3109038       0.0      482.74       0.000  

it's possible to see that the asset number 411 dont have values for the columns ac_voltage and ac_current

In [38]:
df_solar_only_not_na = df_solar[~df_solar["ac_voltage"].isna()]
print(f"Median: {df_solar_only_not_na['ac_current'].median()}")
print(f"Average: {df_solar_only_not_na['ac_current'].mean()}")


Median: 4.679
Average: 11.763000043458105
